### PR TITLE
Allow disabling channels

### LINKED ISSUE #1024: How to disable Nix channels?
I manage my system with flakes, so channels sometimes lead to surprising results.
I tried settings `nix.channel.enable = false` as it is on NixOS but found out that this is not an option.

### PR DESCRIPTION
**Pull request by thecaralice**
Resolves #1024.

### CODE REVIEW

File: modules/environment/default.nix [hunk 1]
```diff
@@ -46,7 +46,8 @@ in
     environment.postBuild = mkOption {
       type = types.lines;
       default = "";
-      description = "Commands to execute when building the global environment.";
+      visible = false;
+      description = "Deprecated alias of {option}`environment.extraSetup`";
```
Comment (Enzime) [inline modules/environment/default.nix:50]: What's the reason behind deprecating `environment.postBuild`?

File: modules/nix/default.nix [hunk 2]
```diff
@@ -768,17 +768,11 @@ in
...
-      (mkIf (config.system.stateVersion > 3) (mkOrder 1200
-      [ { darwin-config = "${config.environment.darwinConfig}"; }
-        "/nix/var/nix/profiles/per-user/root/channels"
-      ]))
-    ];
+    nix.nixPath = mkIf (config.system.stateVersion < 2) (mkDefault [
+      "darwin=$HOME/.nix-defexpr/darwin"
+      "darwin-config=$HOME/.nixpkgs/darwin-configuration.nix"
+      "/nix/var/nix/profiles/per-user/root/channels"
+    ]);
```
Comment (Enzime) [inline modules/nix/default.nix:775]: What does this change do?

File: modules/environment/default.nix (hunk 1 above)
Comment (thecaralice) [reply modules/environment/default.nix:50]: NixOS uses `environment.extraSetup` for that, and having two options doing the same thing is error-prone, so I thought I should introduce the NixOS option and make the old one warn on use.

File: modules/nix/default.nix (hunk 2 above)
Comment (thecaralice) [reply modules/nix/default.nix:775]: See https://github.com/LnL7/nix-darwin/issues/1024#issuecomment-2266718542

File: modules/system/checks.nix [hunk 3]
```diff
@@ -236,7 +236,7 @@ in
 
     system.checks.verifyNixChannels = mkOption {
       type = types.bool;
-      default = true;
+      default = !config.nix.channel.enable;
```
Comment (Enzime) [inline modules/system/checks.nix:239]:
```suggestion
      default = config.nix.channel.enable;
```

Shouldn't this be the other way around?

File: modules/environment/default.nix [hunk 4]
```diff
@@ -206,5 +219,8 @@ in
       ${concatStringsSep "\n" aliasCommands}
     '';
 
+    warnings = lib.optional (
+      cfg.postBuild != ""
+    ) "The option `environment.postBuild` has been renamed to `environment.extraSetup`";
```
Comment (Enzime) [inline modules/environment/default.nix:224]:
```suggestion
```

This should use `mkRenamedOption` instead

File: modules/environment/default.nix [hunk 5]
```diff
@@ -147,6 +148,17 @@ in
       '';
       type = types.lines;
     };
+
+    environment.extraSetup = mkOption {
+      type = types.lines;
+      default = cfg.postBuild;
```
Comment (Enzime) [inline modules/environment/default.nix:154]:
```suggestion
      default = "";
```

File: modules/environment/default.nix [hunk 6]
```diff
@@ -46,7 +46,8 @@ in
     environment.postBuild = mkOption {
```
Comment (Enzime) [inline modules/environment/default.nix:46]:
```suggestion
```

The option should be removed and `mkRenamedOption` should be used instead

**Review by Enzime — CHANGES_REQUESTED** [review]

File: modules/system/checks.nix (hunk 3 above)
Comment (thecaralice) [reply modules/system/checks.nix]: fixed

File: modules/environment/default.nix (hunk 4 above)
Comment (thecaralice) [reply modules/environment/default.nix:224]: Didn't find that in docs, will do

File: modules/environment/default.nix [hunk 7]
```diff
@@ -147,8 +141,23 @@ in
...
+        This should only be used for things that need to modify the internals
+        of the environment, e.g. generating MIME caches.
+        The environment being built can be accessed at $out.
+      '';
+    };
   };
 
+  imports = [
+    (mkRenamedOptionModule ["environment" "postBuild"] ["environment" "extraSetup"])
+  ];
```
Comment (Enzime) [inline modules/environment/default.nix:159]: Can you move this to be before all the options?

**Review by Enzime — COMMENTED** [review]
Everything else looks good to me

File: modules/environment/default.nix (hunk 7 above)
Comment (thecaralice) [reply modules/environment/default.nix]: Done

**Review by Enzime — APPROVED** [review]
LGTM

Thanks for the contribution

**Pull request merged**